pkg/prometheus/operator.go

```diff
@@ -310,6 +311,12 @@ func New(ctx context.Context, conf operator.Config, logger log.Logger, r prometh
 		c.nsPromInf = newNamespaceInformer(c, c.config.Namespaces.PrometheusAllowList)
 	}
 
+	endpointSliceSupported := false
+	_, err = c.kclient.DiscoveryV1beta1().EndpointSlices(kubeletObjectNamespace).List(ctx, metav1.ListOptions{})
```

Comment:
you could also use [k8s.io/client-go/discovery](https://pkg.go.dev/k8s.io/client-go@v0.23.3/discovery). The advantage is that it would fallback to the "legacy" mode if for some reason the API returned an error (e.g. temporary network hiccup).
It would also be worth having an info log reporting whether the API supports the endpointslices API group or not.

Reply:
Ok, I think we can use the discovery serverinfo api to sure if stable endpointslices are supported,ref [doc](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/),WDYT?